lib/taste_tester/commands.rb

```diff
@@ -210,5 +210,200 @@ def self.upload
       logger.error(exception.backtrace.join("\n"))
       exit 1
     end
+
+    def self.impact
+      # Use the repository specified in config.rb to calculate the changes
+      # that may affect Chef. These changes will be further analyzed to
+      # determine specific roles which may change due to modifed dependencies.
+      repo = BetweenMeals::Repo.get(
+        TasteTester::Config.repo_type,
+        TasteTester::Config.repo,
+        logger,
+      )
+      if repo && !repo.exists?
+        fail "Could not open repo from #{TasteTester::Config.repo}"
+      end
+
+      changes = _find_changeset(repo)
+
+      # Use Knife (or custom logic) to check the dependencies of each role
+      # against the list of changes. `impact_roles` will contian the set
+      # of roles with direct or indirect (dependency) modifications.
+      if TasteTester::Config.use_custom_impact_hook
+        impact_roles = TasteTester::Hooks.custom_impact(changes)
```

Comment:
What if we treat this the same we do `print_impact` below.

So we do:

impact_roles = TasteTester::Hooks.custom_impact(changes)
unless impact_roles
  _find_impact(changes)
end
```

Then we don't need these extra flags at all. The default plugin returns nil, so we get the default behavior.

Reply:
Yeah, that was my thought.  Have the "custom_impact" hook (renamed of course) be the actual functionality and users can override it using the same hook mechanism as the post_impact/print functions.